src/test/java/teammates/test/pageobjects/AppPage.java

```diff
@@ -967,18 +966,20 @@ public void verifyContainsElement(By by) {
      * The check is done multiple times with waiting times in between to account for
      * timing issues due to page load, inconsistencies in Selenium API, etc.
      */
-    public void waitForTextsForAllStatusMessagesToUserEquals(final String... expectedTexts) {
-        Assumption.assertFalse(expectedTexts.length == 0);
+    public void waitForTextsForAllStatusMessagesToUserEquals(final String firstExpectedText,
```

Comment:
Is there a particular reason why you think `final` adds a lot of value? There have been already a lot of discussions online about this.

Reply:
To be honest, I did not put too much thought into it. The earlier implementation of `verifyStatus` used `final` (as well as many other methods in `AppPage.java`), so, I thought it was a layer of security to confirm that no developer accidentally tries to assign another value to these variables.

Reply:
I think the unfortunate problem with `final` parameters is that it is also verbose and adds little value.

In any case, we can always enforce no reassignable parameters with static analysis tools and TEAMMATES already does this through PMD :). This way, the developer doesn't have the burden to explicitly add it in every method.

Reply:
Okay. Dropping the `final` then.